autoload/go/util.vim

```diff
@@ -37,6 +37,46 @@ function! go#util#IsWin()
     return 0
 endfunction
 
+"Check if has vimproc
+function! s:has_vimproc()
+    if !exists('g:go#use_vimproc')
+        if go#util#IsWin()
+            try
+                call vimproc#version()
+                let exists_vimproc = 1
+            catch
+                let exists_vimproc = 0
+            endtry
+        else
+            let exists_vimproc = 0
+        endif
+
+        let g:go#use_vimproc = exists_vimproc
+    endif
+
+    return g:go#use_vimproc
+endfunction
+
+if s:has_vimproc()
+    let s:vim_system = get(g:, 'gocomplete#system_function', 'vimproc#system2')
+    let s:vim_shell_error = get(g:, 'gocomplete#shell_error_function', 'vimproc#get_last_status')
+else
+    let s:vim_system = get(g:, 'gocomplete#system_function', 'system')
+    let s:vim_shell_error = ''
+endif
+
+function! go#util#system(str, ...)
+    return call(s:vim_system, [a:str] + a:000)
+endfunction
```

Comment:
Just had time due my new son :) So I just had a new idea and curious about your thought. What if we add a function that  returns both the output and shell error. A list that contains the following `[out, shell_error]` ? This way we could write code like this:

```
let [out, shell_error] = go#util#Run("command ...")
if shell_error != 0 
    return -1
endif

// continue with out
```

This would make it more safer to use and would emulate the way we use it in Go. Let me know what you think. We can do it in another PR btw :)